main.py

```diff
@@ -176,72 +295,19 @@ def main() -> None:
     )
     args = parser.parse_args()
 
-    js = Path(args.js).resolve()
-    test262 = Path(args.test262).resolve()
-
-    print("Searching test files...")
-    if Path(args.pattern).resolve().is_file():
-        test_files = [Path(args.pattern).resolve()]
-    else:
-        test_files = [
-            path.resolve()
-            for path in test262.glob(args.pattern)
-            if path.is_file() and not path.stem.endswith("FIXTURE")
-        ]
-    test_files.sort()
-    total_tests = len(test_files)
-    print(f"Found {total_tests}.")
-
-    remaining_tests = total_tests
-    passed_tests = 0
-    failed_tests = 0
-
+    runner = Runner(args.concurrency, args.timeout)
+    runner.set_interpreter(args.js)
+    runner.find_tests(args.test262, args.pattern)
     start_time = datetime.datetime.now()
-    try:
-        for i, test_file in enumerate(test_files):
-            if args.verbose:
-                print(f"Running test: {test_file}")
-            test_result, output = run_test(js, test262, test_file, timeout=args.timeout)
-            if test_result == TestResult.SUCCESS:
-                passed_tests += 1
-                emoji = "✅"
-            elif test_result == TestResult.FAILURE:
-                failed_tests += 1
-                emoji = "❌"
-            elif test_result == TestResult.NONZERO_EXIT_ERROR:
-                failed_tests += 1
-                emoji = "❗️"
-            elif test_result == TestResult.RUNNER_EXCEPTION:
-                failed_tests += 1
-                emoji = "💥"
-            elif test_result == TestResult.TIMEOUT_ERROR:
-                failed_tests += 1
-                emoji = "💀"
-            elif (
-                test_result == TestResult.METADATA_ERROR
-                or test_result == TestResult.LOAD_ERROR
-            ):
-                failed_tests += 1
-                emoji = "⚠️"
-            else:
-                assert False
-            remaining_tests -= 1
-            progress = int((total_tests - remaining_tests) / total_tests * 100)
-            print(f"[{progress:>3}%] {emoji} {test_file.relative_to(test262)}")
-            if args.verbose and output.strip():
-                print(output)
-    except KeyboardInterrupt:
-        pass
+    runner.run()
     end_time = datetime.datetime.now()
-
-    print("----------------")
-    print(f"Tests: {total_tests}")
-    if remaining_tests:
-        print(f"Remaining: {remaining_tests}")
-    print(f"Passed: {passed_tests}")
-    print(f"Failed: {failed_tests}")
     print(f"Duration: {end_time-start_time}")
```

Comment:
This seems redundant as it's part of the progress bar now, even when finished.

Reply:
You mean the Duration?

Reply:
Yep, looks like we no longer have to keep track of it manually